images/miq-app/docker-assets/container-scripts/container-deploy-common.sh

```diff
@@ -38,52 +29,34 @@ function write_v2_key() {
 KEY
 }
 
-# Inspect PV for previous deployments, if a DB a config is present, restore
-# Source previous deployment info file from PV and compare data with current environment
-# Evaluate conditions and decide a target deployment type: redeploy,upgrade or new
 function check_deployment_status() {
   echo "== Checking deployment status =="
 
-  if [[ -f ${PV_DEPLOY_INFO_FILE} ]]; then
-    echo "== Found existing deployment configuration =="
-    echo "== Restoring existing database configuration =="
-
-    # Source original deployment info variables from PV
-    source ${PV_DEPLOY_INFO_FILE}
-
-    # Obtain current running environment
-    APP_VERSION="$(cat ${APP_ROOT}/VERSION)"
-    SCHEMA_VERSION="$(cd ${APP_ROOT} && RAILS_USE_MEMORY_STORE=true bin/rake db:version | awk '{ print $3 }')"
-    # Check if we have identical EVM versions (exclude master builds)
-    if [[ ${APP_VERSION} == ${PV_APP_VERSION} && ${APP_VERSION} != master ]]; then
-      echo "== App version matches original deployment =="
-      # Check if we have same schema version for same EVM version
-      if [ "${SCHEMA_VERSION}" != "${PV_SCHEMA_VERSION}" ]; then
-        echo "ERROR: Something seems wrong, db schema version mismatch for the same app version: ${PV_SCHEMA_VERSION} <-> ${SCHEMA_VERSION}"
-        exit 1
-      fi
-      # Assuming redeployment (same APP_VERSION)
-      DEPLOYMENT_STATUS=redeployment
-    else
-      # Handle special master case
-      # Master version remains static, check DB schema status and proceed accordingly
+  cd "${APP_ROOT}"
 
-      if [[ ${APP_VERSION} == master ]]; then
-        # Go for redeployment case unless rake task returns 1 (pending migrations)
+  SCHEMA_VERSION="$(RAILS_USE_MEMORY_STORE=true bin/rake db:version | grep "Current version" | awk '{ print $3 }')"
+  echo "Current schema version is $SCHEMA_VERSION"
+  if [ "$SCHEMA_VERSION" == "0" ]; then
```

Comment:
I'm starting to wonder if all of this logic might be better if it was a self-contained rake task.

That way we could cover it with CI.
The only issue I'm not sure about is how to communicate the result.

As @bdunne and I started looking into logging to STDOUT for the aggregated logging effort we had a few issues parsing strings from rake output. Maybe we should define a return code enum of some sort if we go that route.

Thoughts?